### PR TITLE
[CARBONDATA-3294] Fix that MV datamap throw error when using count(1) and case when expression

### PR DESCRIPTION
**Pull request by qiuchenjian**
### [Problem]
MV datamap throw error when using count(1) and case when expression, the error is:
mismatched input 'FROM' expecting {<EOF>, 'WHERE', 'GROUP', 'ORDER', 'HAVING', 'LIMIT', 'LATERAL', 'WINDOW', 'UNION', 'EXCEPT', 'MINUS', 'INTERSECT', 'SORT', 'CLUSTER', 'DISTRIBUTE'}(line 2, pos 0)
== SQL ==
SELECT MT.`3600`, MT.`2250410101`, `countNum`, `rate` 
FROM
^^^

### [Solution]
The compacted SQL has a extra "case when" expression cause this error ,because window operator has a bug when transforming logic plan to modular plan

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/2576/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/10835/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/2806/

File: datamap/mv/core/src/test/scala/org/apache/carbondata/mv/rewrite/MVCountAndCaseTestCase.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                       |   ON sum_result.LAYER4ID = H_REGION.`225040101`
+                       | WHERE H_REGION.NAME_2250410101 IS NOT NULL
+                       | ) MT
+                       | GROUP BY MT.`3600`, MT.`2250410101`
+                       | ORDER BY `3600` ASC LIMIT 5000""".stripMargin)
+
+    assert(verifyMVDataMap(frame.queryExecution.analyzed, "data_table_mv"))
+  }
+
+  def verifyMVDataMap(logicalPlan: LogicalPlan, dataMapName: String): Boolean = {
```
Comment (akashrn5) [inline datamap/mv/core/src/test/scala/org/apache/carbondata/mv/rewrite/MVCountAndCaseTestCase.scala]: `verifyMVDataMap` this method is used in many places, all test cases of MV, so better move ti to some utility class, or some Object and use the same everywhere. Refactor in all the places whoever is using this method.

Comment (qiuchenjian) [reply datamap/mv/core/src/test/scala/org/apache/carbondata/mv/rewrite/MVCountAndCaseTestCase.scala]: @akashrn5.  [PR-3084](https://github.com/apache/carbondata/pull/3084).  handle this, can you help me to merge that PR

**Comment by akashrn5** [comment]
+1

**Comment by qiuchenjian** [comment]
@akashrn5  this pr didn't add any Rule

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**